gui/fitpanelv7/src/RFitPanel6.cxx

```diff
@@ -0,0 +1,512 @@
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <ROOT/RFitPanel6.hxx>
+
+#include <ROOT/RWebWindowsManager.hxx>
+
+#include "TString.h"
+#include "TBackCompFitter.h"
+#include "TGraph.h"
+#include "TROOT.h"
+#include "TF1.h"
+#include "TList.h"
+#include "TPad.h"
+#include "TDirectory.h"
+#include "TBufferJSON.h"
+#include "TMath.h"
+#include "Math/Minimizer.h"
+#include "TColor.h"
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+
+/** \class ROOT::Experimental::RFitPanel
+\ingroup webdisplay
+
+web-based FitPanel prototype.
+*/
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Returns RWebWindow instance, used to display FitPanel
+
+std::shared_ptr<ROOT::Experimental::RWebWindow> ROOT::Experimental::RFitPanel6::GetWindow()
+{
+   if (!fWindow) {
+      fWindow = RWebWindowsManager::Instance()->CreateWindow();
+
+      fWindow->SetPanelName("rootui5.fitpanel.view.FitPanel");
+
+      fWindow->SetDataCallBack([this](unsigned connid, const std::string &arg) { ProcessData(connid, arg); });
+
+      fWindow->SetGeometry(400, 650); // configure predefined geometry
+   }
+
+   return fWindow;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Show FitPanel
+
+void ROOT::Experimental::RFitPanel6::Show(const std::string &where)
+{
+   GetWindow()->Show(where);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Hide FitPanel
+
+void ROOT::Experimental::RFitPanel6::Hide()
+{
+   if (!fWindow)
+      return;
+
+   fWindow->CloseConnections();
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Process data from FitPanel
+/// OpenUI5-based FitPanel sends commands or status changes
+
+void ROOT::Experimental::RFitPanel6::ProcessData(unsigned connid, const std::string &arg)
+{
+   if (arg == "CONN_READY") {
+      fConnId = connid;
+      printf("FitPanel connection established %u\n", fConnId);
+      fWindow->Send(fConnId, "INITDONE");
+      ROOT::Experimental::RFitPanelModel6 model;
+
+      //ComboBox for Data Set
+      if (fHist) {
+         model.fDataSet.emplace_back("0", Form("%s::%s", fHist->ClassName(), fHist->GetName()));
+         model.fSelectDataId = "0";
+      }
+
+      // if (gDirectory) {
+      //    TIter iter(gDirectory->GetList());
+      //    TObject *item = nullptr;
+
+      //    while ((item = iter()) != nullptr)
+      //       if (item->InheritsFrom(TH1::Class()))
+      //          model.fDataSet.emplace_back(item->GetName(), Form("%s::%s", item->ClassName(), item->GetName()));
+      // }
+
+       //ComboBox for Fit Function --- Type
+       model.fTypeFunc.emplace_back("0", "Predef-1D");
```

Comment:
Can this whole part (or at least most of it) move into the `RFitPanelModel6` construction / initialization?

Reply:
I will extract model to extra source file and will put initialization there.